src/components/Shared/Filters/NullState/NullState.test.tsx

```diff
@@ -2,18 +2,48 @@ import React from 'react';
 import { ThemeProvider } from '@mui/material/styles';
 import userEvent from '@testing-library/user-event';
 import { SnackbarProvider } from 'notistack';
+import TestRouter from '__tests__/util/TestRouter';
 import TestWrapper from '__tests__/util/TestWrapper';
 import { render, waitFor } from '__tests__/util/testingLibraryReactMock';
 import { TaskModalEnum } from 'src/components/Task/Modal/TaskModal';
+import { UrlFiltersProvider } from 'src/components/common/UrlFiltersProvider/UrlFiltersProvider';
 import useTaskModal from '../../../../hooks/useTaskModal';
 import theme from '../../../../theme';
 import NullState from './NullState';
 
-const changeFilters = jest.fn();
 const openTaskModal = jest.fn();
 
 jest.mock('src/hooks/useTaskModal');
 
+interface TestComponentProps {
+  page: 'contact' | 'task';
+  totalCount: number;
+  router?: object;
+}
+
+const defaultRouter = {
+  query: {},
+  isReady: true,
+};
+
+const TestComponent: React.FC<TestComponentProps> = ({
+  page,
+  totalCount,
+  router = defaultRouter,
```

Comment:
This is fine how you did it, but since these tests only need to customize the filter, I would have the `TestComponent` accept a `filters` prop. Then `TestComponent` can use it to build the `router` object. I always prefer to put a little more logic in the `TestComponent` and make it a littler smarter if it lets us make the individual tests simpler.